### PR TITLE
feat(api core): simplify from_rfc3339 methods

### PR DESCRIPTION
**Pull request by Emar-Kar**
This PR was reopened for the clearness of the provided changes. (Old one [8742](https://github.com/googleapis/google-cloud-python/pull/8742))

Closes: #7676

### CODE REVIEW

File: api_core/google/api_core/bidi.py [hunk 1]
```diff
@@ -172,7 +172,9 @@ def __init__(self, access_limit, time_window):
 
         self._time_window = time_window
         self._access_limit = access_limit
-        self._past_entries = collections.deque(maxlen=access_limit)  # least recent first
+        self._past_entries = collections.deque(
+            maxlen=access_limit
```
Comment (crwilcox) [inline api_core/google/api_core/bidi.py]: unrelated?

File: api_core/google/api_core/datetime_helpers.py [hunk 2]
```diff
@@ -169,6 +156,27 @@ def from_rfc3339_nanos(value):
     return bare_seconds.replace(microsecond=micros, tzinfo=pytz.utc)
 
 
+def from_rfc3339_nanos(value):
```
Comment (crwilcox) [inline api_core/google/api_core/datetime_helpers.py]: I think we can just have `from_rfc3339_nanos=from_rfc3339`. This isn't typically directly used by users, and keeping it around as an alias isn't going to be a maintenance burden.

File: api_core/google/api_core/protobuf_helpers.py [hunk 3]
```diff
@@ -15,17 +15,19 @@
...
-except ImportError:  # Python 2.7
-    import collections as collections_abc
 import copy
 import inspect
 
 from google.protobuf import field_mask_pb2
 from google.protobuf import message
 from google.protobuf import wrappers_pb2
 
+try:
```
Comment (crwilcox) [inline api_core/google/api_core/protobuf_helpers.py]: This seems out of order? Did a linter do this?

Comment (Emar-Kar) [reply api_core/google/api_core/protobuf_helpers.py]: It was some sort of the `flake8` error and my [build](https://source.cloud.google.com/results/invocations/57246fd0-a187-4c77-be97-85a89b8564b4/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fapi_core) failed a couple of times.
image

File: api_core/google/api_core/bidi.py (hunk 1 above)
Comment (Emar-Kar) [reply api_core/google/api_core/bidi.py]: It was modified during the `black` session.
